src/views/RasterPlot/TimeScrollView/TimeScrollView.tsx

```diff
@@ -147,12 +159,63 @@ export const use2dPanelDataToPixelMatrix = (pixelsPerSecond: number, startTimeSe
 }
 
+export const useTimespanToPixelMatrix = (timeToPixelMatrix: Matrix) => {
+    return useMemo(() => {
+        // Get a matrix that operates on matrix([[starttimesSec], [endtimesSec]]) to yield [[startPixels], [pixelWidths]]
+        const baseTransformValues = timeToPixelMatrix.valueOf().flat()
+        const pixelsPerSecond = baseTransformValues[0]
+        const netPixelOffset = baseTransformValues[1]
+
+        // First row scales the start time and adds offset
+        // Second row subtracts the end time (presumed larger) from the start time, no offset
+        const transform = matrix([
+            [ pixelsPerSecond,      0         ,  netPixelOffset],
+            [-pixelsPerSecond, pixelsPerSecond,       0        ]
+        ])
+        return transform
+    }, [timeToPixelMatrix])
+}
+
 export const getYAxisPixelZero = (TransformMatrix2d: Matrix) => {
     const augmentedZero = matrix([[0], [0], [1]])
     const value = (multiply(TransformMatrix2d, augmentedZero).valueOf() as number[][])[1][0]
     return value
 }
 
+export const convertStartAndEndTimesToPixelRects = (startsSec: number[], endsSec: number[], timespanTransform: Matrix) => {
+    const augmentedTimes = matrix([startsSec, endsSec, new Array(startsSec.length).fill(1)])
+    const pixelStartsAndWidths = multiply(timespanTransform, augmentedTimes).valueOf()
+    // returns array w/ first index = list of start times (in pixels), second index = list of widths (in pixels)
+    return pixelStartsAndWidths as number[][]
+}
+
+export const filterTimeRanges = (startsSec: number[], endsSec: number[], rangeStartSec: number | undefined, rangeEndSec: number | undefined): number[][] => {
+    // Take all intervals that begin before the range end or end after the range start
+    const definedRangeStartSec = rangeStartSec || 0
+
+    const startIndices = (rangeEndSec ? startsSec.map((s, i) => s <= rangeEndSec ? i : null) : startsSec.map((s, i) => i)).filter(i => i) as number[] // filter removes nulls
+    const endIndices = endsSec.map((s, i) => s >= definedRangeStartSec ? i : null).filter(i => i) as number[]
+    // get set intersection to find all those spans that begin before our window ends and end after our window begins
+    const largerSet = new Set(startIndices.length > endIndices.length ? startIndices : endIndices)
+    const smallerList = startIndices.length > endIndices.length ? endIndices : startIndices
+    const validIndices = smallerList.filter(i => largerSet.has(i))
+    const finalStarts = validIndices.map(i => Math.max(startsSec[i], definedRangeStartSec))
+    const finalEnds = validIndices.map(i => Math.min(endsSec[i], (rangeEndSec ?? endsSec[i])))
+    return [finalStarts, finalEnds]
+}
+
+export const filterAndProjectHighlightSpans = (highlightIntervals: HighlightIntervalSet[], visibleTimeStartSeconds: number | undefined, visibleTimeEndSeconds: number | undefined, spanTransform: Matrix) => {
```

Comment:
This is probably the main entry point for other components that want to use this overlay. They basically just need to provide data that came from other hooks & the actual sets of intervals that need to be highlighted.

This functionality (like the filtering functionality) is centralized here, close to where it's used.